src/wallet/rpcwallet.cpp

```diff
@@ -2977,6 +3119,126 @@ UniValue z_getbalance(const UniValue& params, bool fHelp)
+            "\nArguments:\n"
+            "1. \"address\"      (string) The selected address. It may be a transparent or private address.\n"
+            "2. minconf        (numeric, optional, default=1) Only include transactions confirmed at least this many times.\n"
+            "\nResult:\n"
+            "{\n"
+            "  \"pools\": {\n"
+            "    \"transparent\": {\n"
+            "        \"valueZat\": amount   (numeric) The amount held in the transparent pool by this address\n"
+            "    \"},\n"
+            "    \"sprout\": {\n"
+            "        \"valueZat\": amount   (numeric) The amount held in the sprout pool by this address\n"
+            "    \"},\n"
+            "    \"sapling\": {\n"
+            "        \"valueZat\": amount   (numeric) The amount held in the sapling pool by this address\n"
+            "    \"},\n"
+            "    \"orchard\": {\n"
+            "        \"valueZat\": amount   (numeric) The amount held in the orchard pool by this address\n"
+            "    \"}\n"
+            "  \"}\n"
+            "}\n"
+            "Result amounts are in units of " + MINOR_CURRENCY_UNIT + ".\n"
+            "Pools for which the balance is zero are not shown.\n"
+            "\nExamples:\n"
+            "\nThe per-pool amount received by address \"myaddress\" with at least 1 block confirmed\n"
+            + HelpExampleCli("z_getbalanceforaddress", "\"myaddress\"") +
+            "\nThe per-pool amount received by address \"myaddress\" with at least 5 blocks confirmed\n"
+            + HelpExampleCli("z_getbalanceforaddress", "\"myaddress\" 5") +
+            "\nAs a json rpc call\n"
+            + HelpExampleRpc("z_getbalanceforaddress", "\"myaddress\", 5")
+        );
+
+    if (!fExperimentalOrchardWallet) {
+        throw JSONRPCError(RPC_WALLET_ENCRYPTION_FAILED, "Error: the Orchard wallet experimental extensions are disabled.");
+    }
+    std::string fromaddress = params[0].get_str();
+    UniValue pools(UniValue::VOBJ);
+    pools.pushKV("transparent", 99999.99);
+    pools.pushKV("sprout", 99999.99);
+    pools.pushKV("sapling", 99999.99);
+    pools.pushKV("orchard", 99999.99);
+
+    UniValue result(UniValue::VOBJ);
+    result.pushKV("pools", pools);
+
+    return result;
+}
+
+UniValue z_getbalanceforaccount(const UniValue& params, bool fHelp)
+{
+    if (!EnsureWalletIsAvailable(fHelp))
+        return NullUniValue;
+
+    if (fHelp || params.size() < 1 || params.size() > 2)
+        throw runtime_error(
+            "z_getbalanceforaccount account ( minconf )\n"
+            "\nReturns the pool balances of the given account."
+            "\nArguments:\n"
+            "1. account      (numeric) The account number.\n"
+            "2. minconf      (numeric, optional, default=1) Only include transactions confirmed at least this many times.\n"
+            "\nResult:\n"
+            "{\n"
+            "  \"pools\": {\n"
```

Comment:
Add the account number (and maybe a `"minimumConfirmations"` value) to the output?
```suggestion
            "  \"protocolBalances\": {\n"
```